internal/service/synthetics/canary.go

```diff
@@ -388,7 +394,9 @@ func resourceCanaryRead(ctx context.Context, d *schema.ResourceData, meta any) d
 	}.String()
 	d.Set(names.AttrARN, canaryArn)
 	d.Set("artifact_s3_location", canary.ArtifactS3Location)
-	d.Set("engine_arn", canary.EngineArn)
+	if len(canary.EngineConfigs) > 0 {
+		d.Set("engine_arn", canary.EngineConfigs[0].EngineArn)
+	}
```

Comment:
```suggestion
	if len(canary.EngineConfigs) > 0 {
		d.Set("engine_arn", canary.EngineConfigs[0].EngineArn)
	} else {
		d.Set("engine_arn", canary.EngineArn)
	}
```
Backwards compatibility.